### PR TITLE
Makes causal lm metrics configurable.

### PR DESCRIPTION
**Pull request by markblee**

### CODE REVIEW

**Review by ruomingp — CHANGES_REQUESTED** [review]
Please see my comment on PredictModel in the internal PR.

**Comment by markblee** [comment]
> Please see my comment on PredictModel in the internal PR.

Thanks. Removed as it's not critical to the PR.

**Review by ruomingp — APPROVED** [review]

File: axlearn/common/causal_lm.py [hunk 1]
```diff
@@ -32,16 +32,242 @@
...
+        target_num_bytes: Optional[Tensor] = input_batch.get("target_num_bytes")
+        logits = predict_outputs["logits"]
+
+        live_targets = target_labels >= 0
+        num_targets = live_targets.sum()
+        accuracy = (
+            jnp.equal(jnp.argmax(logits, axis=-1), target_labels) * live_targets
+        ).sum() / jnp.maximum(1, num_targets)
+
+        loss, loss_dict = cross_entropy(
```
Comment (apivovarov) [inline axlearn/common/causal_lm.py:103]: Hi Mark, do we need a separate accuracy calculation here (above cross_entropy) if the loss_dict from cross_entropy already contains an accuracy entry? @markblee
